rtl/eclrtl/rtldynfield.cpp

```diff
@@ -1363,47 +1447,50 @@ class GeneralRecordTranslator : public CInterfaceOf<IDynamicTransform>
+        case type_int:
+        case type_swapint:
+        case type_packedint:
+        case type_filepos:
+        case type_keyedint:
+        {
+            __int64 res = rtlStrToInt8(srcSize, source);
+            offset = destType.buildInt(builder, offset, field, res);
+            break;
         }
-        if (matchFlags & ~(match_perfect|match_link|match_none|match_virtual|match_extend|match_truncate))
+        case type_real:
         {
-            for (unsigned idx = 0; idx < destRecInfo.getNumFields(); idx++)
-            {
-                const MatchInfo &match = matchInfo[idx];
-                const RtlTypeInfo *type = destRecInfo.queryType(idx);
-                unsigned matchField = match.matchIdx;
-                switch (match.matchType)
-                {
-                case match_perfect:
-                case match_link:
-                case match_none:
-                case match_virtual:
-                case match_extend:
-                case match_truncate:
-                    // These ones were already included in fixedDelta
-                    break;
-                default:
-                    // This errs on the side of small - i.e. it assumes that all typecasts end up at minimum size
-                    // We could do better in some cases e.g. variable string <-> variable unicode we can assume factor of 2,
-                    // uft8 <-> string we could calculate here - but unlikely to be worth the effort.
-                    // But it's fine for fixed size output fields, including truncate/extend
-                    // We could also precalculate the expected delta if all omitted fields are fixed size - but not sure how likely/worthwhile that is.
-                    expectedSize += type->getMinSize() - sourceRow.getSize(matchField);
-                    //DBGLOG("Adjusting estimated size by (%d - %d) to %d for translated field %d (%s)", (int) sourceRow.getSize(matchField), type->getMinSize(), expectedSize, matchField, sourceRecInfo.queryName(matchField));
-                    break;
-                }
-            }
+            double res = rtlStrToReal(srcSize, source);
+            offset = destType.buildReal(builder, offset, field, res);
+            break;
         }
-        return expectedSize;
+        case type_data:
+        case type_string:
+        case type_decimal:  // Go via string - not common enough to special-case
+        case type_varstring:
+        case type_qstring:
+        case type_utf8:
+            //MORE: Could special case casting from utf8 to utf8 similar to strings above
+        case type_unicode:
+        case type_varunicode:
+        {
+            size32_t utf8chars = rtlUtf8Length(srcSize, source);
+            offset = destType.buildUtf8(builder, offset, field, utf8chars, source);
+            break;
+        }
+        case type_set:
+        {
+            UNIMPLEMENTED; // JCS->GH - but perhaps can/should translate using iterator too?
```

Comment:
Is there a Jira to add this functionality? Is it needed?

Reply:
Gavin thinks could be implemented, so will leave comment in but have opened a JIRA:
https://track.hpccsystems.com/projects/HPCC/issues/HPCC-22323